### ISSUE TITLE
Nested-nested queries not working

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#5482]
According to @mazameli there seems to be some bugs with using nested queries inside nested queries.

**Comment by camsaul** [comment]
Also need to update the docs and take out the part where it says you can't use nested-nested queries once this is fixed

**Comment by joshuashort** [comment]
I don't remember if I opened an issue about this or if I just mentioned it on the Discourse site, but I've found that nested queries don't work when there are custom fields defined in the base query..

For example, I have a query that defines some custom fields and filters on those fields -- when I try to run the simplest possible nested query (i.e. a pass-through, without any extra customizations) I get the following:

![screen shot 2017-07-14 at 4 28 47 pm](https://user-images.githubusercontent.com/1498607/28234212-0e03a4d2-68b2-11e7-82d4-a2b85f17abee.png)

There are only 2 custom fields defined, both of which are visible in the base query filter:

![screen shot 2017-07-14 at 4 29 09 pm](https://user-images.githubusercontent.com/1498607/28234228-3377b762-68b2-11e7-9e87-78ac161e1173.png)

Here are some log messages for fun:

```
07-14 23:34:54 DEBUG middleware.permissions :: Permissions Check 🔒 : Can User 1 access Table 26 (hospital_case_info)?
07-14 23:34:54 DEBUG middleware.permissions :: Permissions Check 🔓 : Yes ✅  because User 1 is a member of Group 2 (Administrators) which has permissions for '/'
07-14 23:34:54 WARN metabase.query-processor :: {:status :failed,
 :class java.lang.Exception,
 :error "No expression named 'Injury to Surgery Time'.",
 :stacktrace
 ["driver.generic_sql.query_processor$expression_with_name.invokeStatic(query_processor.clj:56)"
  "driver.generic_sql.query_processor$expression_with_name.invoke(query_processor.clj:52)"
  "driver.generic_sql.query_processor$fn__41628.invokeStatic(query_processor.clj:89)"
  "driver.generic_sql.query_processor$fn__41628.invoke(query_processor.clj:74)"
  "driver.generic_sql.query_processor$fn__41614$G__41609__41619.invoke(query_processor.clj:70)"
  "driver.generic_sql.query_processor$filter_subclause__GT_predicate.invokeStatic(query_processor.clj:206)"
  "driver.generic_sql.query_processor$filter_subclause__GT_predicate.invoke(query_processor.clj:202)"
  "driver.generic_sql.query_processor$filter_clause__GT_predicate.invokeStatic(query_processor.clj:227)"
  "driver.generic_sql.query_processor$filter_clause__GT_predicate.invoke(query_processor.clj:219)"
  "driver.generic_sql.query_processor$filter_clause__GT_predicate.invokeStatic(query_processor.clj:224)"
  "driver.generic_sql.query_processor$filter_clause__GT_predicate.invoke(query_processor.clj:219)"
  "driver.generic_sql.query_processor$apply_filter.invokeStatic(query_processor.clj:232)"
  "driver.generic_sql.query_processor$apply_filter.invoke(query_processor.clj:229)"
  "driver.generic_sql$fn__41180$G__41031__41189.invoke(generic_sql.clj:29)"
  "driver.generic_sql.query_processor$apply_clauses.invokeStatic(query_processor.clj:305)"
  "driver.generic_sql.query_processor$apply_clauses.invoke(query_processor.clj:300)"
  "driver.generic_sql.query_processor$apply_source_query$fn__41789.invoke(query_processor.clj:281)"
  "driver.generic_sql.query_processor$apply_source_query.invokeStatic(query_processor.clj:280)"
  "driver.generic_sql.query_processor$apply_source_query.invoke(query_processor.clj:275)"
  "driver.generic_sql.query_processor$apply_clauses.invokeStatic(query_processor.clj:305)"
  "driver.generic_sql.query_processor$apply_clauses.invoke(query_processor.clj:300)"
  "driver.generic_sql.query_processor$build_honeysql_form.invokeStatic(query_processor.clj:317)"
  "driver.generic_sql.query_processor$build_honeysql_form.invoke(query_processor.clj:313)"
  "driver.generic_sql.query_processor$mbql__GT_native.invokeStatic(query_processor.clj:325)"
  "driver.generic_sql.query_processor$mbql__GT_native.invoke(query_processor.clj:321)"
  "driver$fn__22497$G__22375__22504.invoke(driver.clj:45)"
  "query_processor.middleware.mbql_to_native$query__GT_native_form.invokeStatic(mbql_to_native.clj:17)"
  "query_processor.middleware.mbql_to_native$query__GT_native_form.invoke(mbql_to_native.clj:12)"
  "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__26107.invoke(mbql_to_native.clj:26)"
  "query_processor.middleware.annotate_and_sort$annotate_and_sort$fn__24637.invoke(annotate_and_sort.clj:41)"
  "query_processor.middleware.limit$limit$fn__26062.invoke(limit.clj:14)"
  "query_processor.middleware.cumulative_aggregations$cumulative_aggregation$fn__25910.invoke(cumulative_aggregations.clj:46)"
  "query_processor.middleware.cumulative_aggregations$cumulative_aggregation$fn__25910.invoke(cumulative_aggregations.clj:46)"
  "query_processor.middleware.format_rows$format_rows$fn__26053.invoke(format_rows.clj:21)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__27252.invoke(results_metadata.clj:88)"
  "query_processor.middleware.add_row_count_and_status$add_row_count_and_status$fn__24334.invoke(add_row_count_and_status.clj:14)"
  "query_processor.middleware.driver_specific$process_query_in_context$fn__25930.invoke(driver_specific.clj:12)"
  "query_processor.middleware.resolve_driver$resolve_driver$fn__27262.invoke(resolve_driver.clj:14)"
  "query_processor.middleware.cache$maybe_return_cached_results$fn__24717.invoke(cache.clj:146)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__25852.invoke(catch_exceptions.clj:51)"
  "query_processor$process_query.invokeStatic(query_processor.clj:118)"
  "query_processor$process_query.invoke(query_processor.clj:114)"
  "query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:224)"
  "query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:219)"
  "query_processor$fn__27296$process_query_and_save_execution_BANG___27301$fn__27302.invoke(query_processor.clj:262)"
  "query_processor$fn__27296$process_query_and_save_execution_BANG___27301.invoke(query_processor.clj:248)"
  "api.dataset$fn__29098$fn__29101.invoke(dataset.clj:62)"
  "api.common.internal$do_with_caught_api_exceptions.invokeStatic(internal.clj:227)"
  "api.common.internal$do_with_caught_api_exceptions.invoke(internal.clj:222)"
  "api.dataset$fn__29098.invokeStatic(dataset.clj:53)"
  "api.dataset$fn__29098.invoke(dataset.clj:53)"
  "middleware$enforce_authentication$fn__28487.invoke(middleware.clj:122)"
  "api.routes$fn__39589.invokeStatic(routes.clj:60)"
  "api.routes$fn__39589.invoke(routes.clj:60)"
  "routes$fn__40230$fn__40231.doInvoke(routes.clj:75)"
  "routes$fn__40230.invokeStatic(routes.clj:71)"
  "routes$fn__40230.invoke(routes.clj:71)"
  "middleware$log_api_call$fn__28586$fn__28588.invoke(middleware.clj:331)"
  "middleware$log_api_call$fn__28586.invoke(middleware.clj:330)"
  "middleware$add_security_headers$fn__28536.invoke(middleware.clj:245)"
  "middleware$bind_current_user$fn__28491.invoke(middleware.clj:142)"
  "middleware$maybe_set_site_url$fn__28540.invoke(middleware.clj:268)"],
 :query
 {:type "query",
  :query {:source_table "card__11"},
  :parameters [],
  :constraints {:max-results 10000, :max-results-bare-rows 2000},
  :info
  {:executed-by 1,
   :context :ad-hoc,
   :card-id 11,
   :nested? true,
   :query-hash [44, -54, 88, -96, -66, -55, -59, -79, 14, -16, 46, -37, 91, 51, 83, -17, -87, -31, -105, 77, -61, 93, -126, -102, -118, 61, 102, -128, 70, -126, 92, -22],
   :query-type "MBQL"}},
 :expanded-query nil}

07-14 23:34:54 WARN metabase.query-processor :: Query failure: No expression named 'Injury to Surgery Time'.
["query_processor$assert_query_status_successful.invokeStatic(query_processor.clj:193)"
 "query_processor$assert_query_status_successful.invoke(query_processor.clj:186)"
 "query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:225)"
 "query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:219)"
 "query_processor$fn__27296$process_query_and_save_execution_BANG___27301$fn__27302.invoke(query_processor.clj:262)"
 "query_processor$fn__27296$process_query_and_save_execution_BANG___27301.invoke(query_processor.clj:248)"
 "api.dataset$fn__29098$fn__29101.invoke(dataset.clj:62)"
 "api.common.internal$do_with_caught_api_exceptions.invokeStatic(internal.clj:227)"
 "api.common.internal$do_with_caught_api_exceptions.invoke(internal.clj:222)"
 "api.dataset$fn__29098.invokeStatic(dataset.clj:53)"
 "api.dataset$fn__29098.invoke(dataset.clj:53)"
 "middleware$enforce_authentication$fn__28487.invoke(middleware.clj:122)"
 "api.routes$fn__39589.invokeStatic(routes.clj:60)"
 "api.routes$fn__39589.invoke(routes.clj:60)"
 "routes$fn__40230$fn__40231.doInvoke(routes.clj:75)"
 "routes$fn__40230.invokeStatic(routes.clj:71)"
 "routes$fn__40230.invoke(routes.clj:71)"
 "middleware$log_api_call$fn__28586$fn__28588.invoke(middleware.clj:331)"
 "middleware$log_api_call$fn__28586.invoke(middleware.clj:330)"
 "middleware$add_security_headers$fn__28536.invoke(middleware.clj:245)"
 "middleware$bind_current_user$fn__28491.invoke(middleware.clj:142)"
 "middleware$maybe_set_site_url$fn__28540.invoke(middleware.clj:268)"]

07-14 23:34:54 DEBUG metabase.middleware :: POST /api/dataset 200 (1 s) (14 DB calls)
```

The salient part is how the engine can't find 'Injury to Surgery Time' in the nested query..

**Comment by salsakran** [comment]
bumping to p1 since we now have a notebook that makes 3 deep nested queries trivial to produce

**Comment by camsaul** [comment]
This has been fixed since at least 0.33.0, possibly sooner
